packages/mobile/src/send/utils.ts

```diff
@@ -192,21 +193,24 @@ export function showLimitReachedError(
 
 export function* handleSendPaymentData(
   data: UriData,
-  cachedRecipient?: RecipientWithContact,
+  cachedRecipient?: Recipient,
   isOutgoingPaymentRequest?: true,
   isFromScan?: true
 ) {
-  const recipient: RecipientWithQrCode = {
-    kind: RecipientKind.QrCode,
+  const recipient: AddressRecipient = {
     address: data.address.toLowerCase(),
-    displayId: data.e164PhoneNumber,
-    displayName: data.displayName || cachedRecipient?.displayName || 'anonymous',
+    name: data.displayName || cachedRecipient?.name || 'anonymous',
```

Comment:
I realize this was already here but lets not hardcode the `anonymous` string, can we just leave it as undefined if there's no name so it defaults to the address in the feed?